euphoria-core/src/main/java/cz/seznam/euphoria/core/util/ExceptionUtils.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016-2017 Seznam.cz, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cz.seznam.euphoria.core.util;
+
+public class ExceptionUtils {
+
+  public static <T> T unchecked(SupplyingAction<T> action) {
+    try {
+      return action.perform();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static <T> void unchecked(Action<T> action) {
+    try {
+      action.perform();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public interface SupplyingAction<T> {
```

Comment:
Also naming seems a little unclear. Could this be `Factory` or just `Supplier`? For better usability it could be handy if this would extend `Serializable`.